promql/engine.go

```diff
@@ -482,13 +482,14 @@ func (ng *Engine) populateSeries(ctx context.Context, q storage.Queryable, s *Ev
 	Inspect(s.Expr, func(node Node, path []Node) error {
 		var set storage.SeriesSet
 		params := &storage.SelectParams{
-			Start: timestamp.FromTime(s.Start.Add(-LookbackDelta)),
+			Start: timestamp.FromTime(s.Start),
 			End:   timestamp.FromTime(s.End),
 			Step:  int64(s.Interval / time.Millisecond),
 		}
 
 		switch n := node.(type) {
 		case *VectorSelector:
+			params.Start = params.Start - durationMilliseconds(LookbackDelta)
```

Comment:
We should probably be setting end here too for consistency

Reply:
As it stands how start/end/step are all set before the switch, and then the attributes that change are set below. I think it could definitely make sense to move the Start down there, but the End is the same regardless of type (vector or matrix). So IMO we either leave the shared initialization above, or make each type switch have the entire creation local to the switch case.

Reply:
Ah I see it now, that's what I get for reviewing code just before bed.